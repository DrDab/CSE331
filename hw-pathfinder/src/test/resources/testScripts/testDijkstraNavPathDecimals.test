# Test if Dijkstra implementation provides correct output navigating complex graph with non-integer distances.

# Create graph g
CreateGraph g

# Add city names to graph g
AddNode g LosSantos
AddNode g SanAndreas
AddNode g SanFierro
AddNode g LasVenturas
AddNode g Ludendorff
AddNode g Vice
AddNode g Liberty
AddNode g Alderney

# Add edges to graph g
AddEdge g LosSantos SanAndreas 0.5
AddEdge g LosSantos SanFierro 3.0
AddEdge g SanAndreas LosSantos 0.05
AddEdge g SanAndreas SanFierro 3.0
AddEdge g SanFierro Ludendorff 150.5
AddEdge g Ludendorff Alderney 2000.24
AddEdge g Alderney Liberty 30.36

# Test pathfinding
FindPath g LosSantos Liberty
