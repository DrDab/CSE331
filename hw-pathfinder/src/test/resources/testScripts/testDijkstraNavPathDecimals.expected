# Test if Dijkstra implementation provides correct output navigating complex graph with non-integer distances.

# Create graph g
created graph g

# Add city names to graph g
added node LosSantos to g
added node SanAndreas to g
added node SanFierro to g
added node LasVenturas to g
added node Ludendorff to g
added node Vice to g
added node Liberty to g
added node Alderney to g

# Add edges to graph g
added edge 0.500 from LosSantos to SanAndreas in g
added edge 3.000 from LosSantos to SanFierro in g
added edge 0.050 from SanAndreas to LosSantos in g
added edge 3.000 from SanAndreas to SanFierro in g
added edge 150.500 from SanFierro to Ludendorff in g
added edge 2000.240 from Ludendorff to Alderney in g
added edge 30.360 from Alderney to Liberty in g

# Test pathfinding
path from LosSantos to Liberty:
LosSantos to SanFierro with weight 3.000
SanFierro to Ludendorff with weight 150.500
Ludendorff to Alderney with weight 2000.240
Alderney to Liberty with weight 30.360
total cost: 2184.100